macros/src/parser/input_state.rs

```diff
@@ -0,0 +1,153 @@
+use syn::{parenthesized, parse, spanned::Spanned, token, Ident, Token, Type};
+
+#[derive(Debug, Clone)]
+pub struct InputState {
+    pub start: bool,
+    pub wildcard: bool,
+    pub ident: Ident,
+    pub data_type: Option<Type>,
+}
+
+impl parse::Parse for InputState {
+    fn parse(input: parse::ParseStream) -> syn::Result<Self> {
+        // Check for starting state definition
+        let start = input.parse::<Token![*]>().is_ok();
+
+        // check to see if this is a wildcard state, which is denoted with "underscore"
+        let underscore = input.parse::<Token![_]>();
+        let wildcard = underscore.is_ok();
+
+        // Input State
+        let ident: Ident = if let Ok(underscore) = underscore {
```

Comment:
What happens if both start and wildcard are used together? It looks like it's allowed.

```
*_ + ...
```

Reply:
Good catch.  This should not be allowed.  I added code to panic if this occurs with an appropriate message, as well as an additional unit test to cover.  I also updated the ReadMe.